syncer/syncer.go

```diff
@@ -272,6 +278,16 @@ func (s *syncer) bulkSyncWithPeer(peerID peer.ID, peerLatestBlock uint64,
 	}
 }
 
+// setIsSyncing updates the isSyncing field
+func (s *syncer) setIsSyncing(isSyncing bool) {
+	s.isSyncing.Store(isSyncing)
+}
+
+// IsSyncingWithPeer indicates if node is syncing with peer
+func (s *syncer) IsSyncingWithPeer() bool {
+	return s.isSyncing.Load()
```

Comment:
Double check, but I believe this can be changed this way (sync progression should be present only if node is currently syncing). If it the case, then we should remove `isSyncing` field.

```suggestion
	return s.GetSyncProgression() != nil
```

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/2020/commits/ad5709292e39658ccf06dfaabaa479800cbcb985